### PR TITLE
[CBRD-24615] Result value error when using query cache in partition table

### PR DESCRIPTION
**Pull request by beyondykk9**
http://jira.cubrid.org/browse/CBRD-24615

When using a partition table, the existing query cache does not disappear even if data is inserted or modified.

Even if it looks at the actual ";info qcache", the existing cache should disappear when data is inserted, but it remains and the previous value is retrieved.

The operation without partitioning is good.

### CODE REVIEW

File: src/transaction/locator_sr.c [hunk 1]
```diff
@@ -5329,8 +5333,12 @@ locator_update_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oid, OID
   REPL_INFO repl_info;
   TDE_ALGORITHM tde_algo = TDE_ALGORITHM_NONE;
 
+  OID superclass_oid;
+
   assert (class_oid != NULL && !OID_ISNULL (class_oid));
 
+  OID_SET_NULL (&superclass_oid);
```
Comment (hgryoo) [inline src/transaction/locator_sr.c:5340]: For information only
the following is the same.
```
OID superclass_oid = OID_INITIALIZER;
```

**Review by hgryoo — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

File: src/transaction/locator_sr.c [hunk 2]
```diff
@@ -6274,12 +6285,35 @@ locator_delete_force_internal (THREAD_ENTRY * thread_p, HFID * hfid, OID * oid,
...
 			    class_oid.pageid, class_oid.slotid, class_oid.volid);
 	    }
+
+	  OID_SET_NULL (&superclass_oid);
+
+	  HFID_COPY (&real_hfid, hfid);
+	  COPY_OID (&real_class_oid, &class_oid);
+
+	  error_code =
+	    partition_prune_update (thread_p, &class_oid, &copy_recdes, NULL, DB_PARTITIONED_CLASS, &real_class_oid,
```
Comment (shparkcubrid) [inline src/transaction/locator_sr.c:6305]: Can't the routine be executed only in the case of a partition table? like below.
```
locator_update_force()
if (pruning_type != DB_NOT_PARTITIONED_CLASS)
```

Comment (beyondykk9) [reply src/transaction/locator_sr.c:6305]: the pruning_type is necessary but the caller doesn't transfer the value. the partition_load_pruning_context called by partition_prun_update  returns null as superclass_oid when the pruning_type is not DB_PARTITION_CLASS. To modify the function call locator_delete_force with pruning_type as an argument needs to modify several routines.
In addition, partition_prune_update is only called at QUERY_CACHE is enabled. Therefore the partition_prune_update 
 calling cost is limited.

File: src/transaction/locator_sr.c (hunk 1 above)
Comment (beyondykk9) [reply src/transaction/locator_sr.c:5340]: Ok, thanks.

**Review by hyunikn — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

File: src/transaction/locator_sr.c [hunk 3]
```diff
@@ -6274,12 +6285,35 @@ locator_delete_force_internal (THREAD_ENTRY * thread_p, HFID * hfid, OID * oid,
       /* remove query result cache entries which are relevant with this class */
       if (!QFILE_IS_LIST_CACHE_DISABLED)
 	{
+	  OID superclass_oid;
```
Comment (ctshim) [inline src/transaction/locator_sr.c:6288]: Please apply hgryoo's comments here as well

OID superclass_oid = OID_INITIALIZER;

**Review by ctshim — APPROVED** [review]

File: src/transaction/locator_sr.c (hunk 3 above)
Comment (beyondykk9) [reply src/transaction/locator_sr.c:6288]: OID_SET_NULL is better readability?

**Review by youngjinj — APPROVED** [review]

**Pull request merged**